Neos.Fusion/Classes/Core/Parser.php

```diff
@@ -50,20 +50,33 @@ class Parser implements ParserInterface
      * Parses the given Fusion source code, resolves includes and returns a merged array tree
      * as the result.
      *
-     * @param string $sourceCode The Fusion source code to parse
-     * @param string|null $contextPathAndFilename An optional path and filename used for relative Fusion file includes
+     * @param string|FusionCodeCollection $sourceCode string is DEPRECATED - The Fusion source code to parse
+     * @param string|null $contextPathAndFilename DEPRECATED - An optional path and filename used for relative Fusion file includes
      * @param array $mergedArrayTreeUntilNow Used internally for keeping track of the built merged array tree
      * @return array The merged array tree for the Fusion runtime, generated from the source code
      * @throws Fusion\Exception
      * @api
      */
-    public function parse(string $sourceCode, ?string $contextPathAndFilename = null, array $mergedArrayTreeUntilNow = []): array
+    public function parse(string|FusionCodeCollection $sourceCode, ?string $contextPathAndFilename = null, array $mergedArrayTreeUntilNow = []): array
```

Comment:
allow string|FusionCode|FusionCodeCollection $sourceCode,
mark deprecations with @deprecated

Reply:
Alternatively add new method `parseFrom` and deprecate the whole parse method